_posts/2017-08-18-comment-verifier-l-orthographe-de-vos-docs-depuis-travis-ci.md

```diff
@@ -0,0 +1,134 @@
+authors:
+    - charles-eric
+date: '2017-08-18 17:00:00 +0100'
+date_gmt: '2017-08-18 16:00:00 +0100'
+categories:
+    - CI
+tags:
+    - CI
+    - Travis
+    - Aspell
+    - Spelling
+cover: /assets/2017-08-18-how-to-check-the-spelling-of-your-docs-from-travis-ci/typing.jpg
+---
+
+Notre blog est basé sur Jekyll et hébergé sur Github Pages : [plus de détails ici](/fr/migration-du-blog/). Donc pour publier un article de blog, chaque auteur doit ouvrir une pull request sur Github pour soumettre ses fichiers markdown.
+Ensuite les autres astronautes peuvent relire ce qui a été rédigé avant de merger, i.e. publier l'article. Bien-sûr le but de cette revue est de s'assurer que tout est bien expliqué et pas uniquement de corriger toutes les fautes d'orthographe, sinon cette relecture ne serait pas très drôle ! ;)
+
+C'est pourquoi nous avions besoin d'un moyen simple permettant de trouver automatiquement toutes les fautes dans les fichiers changés des pull requests, pour faciliter cette revue. Bien-sûr nous savons que les outils de vérification automatique d'orthographe ne sont jamais parfaits, et nous voulions donc seulement que cet outil nous envoie une notification avec les éventuelles erreurs, sans bloquer les autres astronautes qui voudraient merger la pull request.
+
+Voilà ce que nous avons donc fait :
+
+Comment fonctionne Aspell ?
+===========================
+
+Premièrement il faut installer cet outil :
+
+```bash
+apt-get install aspell aspell-en # et aspell-fr pour l'orthographe française par exemple
+```
+
+La commande que nous allons utiliser est la suivante :
+
+```bash
+cat some_text.md | aspell --lang=en --encoding=utf-8 list
+```
+
+- `some_text.md` correspond à votre fichier markdown, dans lequel vous voulez vérifier l'orthographe
+- `--lang=en` ou `--lang=fr` par exemple, en fonction de votre langue
+- `--encoding=utf-8` si votre fichier contient des caractères spéciaux
+
+Cette commande va retourner tous les mots qu'Aspell ne connaît pas, c'est à dire non listés dans ses dictionnaires : nos potentielles fautes d'orthographe.
+
+A noter qu'il peut être intéressant d'exécuter cette commande plusieurs fois, en anglais et en français par exemple, surtout si vous rédigez des documents techniques en français qui contiendront forcément aussi beaucoup de termes anglais.
+
+Bien-sûr vous allez aussi avoir besoin d'autoriser vos propres expressions, qui ne seraient pas reconnues par défaut. Pour cela, vous pouvez ajouter et utiliser vos dictionnaires personnels, nommé `.aspell.en.pws` pour un dictionnaire personnel anglais. Voici un exemple de ce que pourrait contenir ce fichier :
+
+```
+personal_ws-1.1 fr 3
+php
+javascript
+android
+```
+
+A noter que l'entête de ce fichier (première ligne) est importante : les deux derniers arguments correspondent à la langue et au nombre de mots.
+
+Ensuite pour utiliser ce dictionnaire personnel, vous devez ajouter cet argument à la commande : `--personal=./.aspell.en.pws`
```

Comment:
I have a question (maybe reader could have this too): is it possible to add multiple `--personal` options here if we have multiple dictionnaries we want to use? or something like `--personal=./.aspell.*.pws`?